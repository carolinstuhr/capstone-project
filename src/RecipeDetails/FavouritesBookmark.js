import React from 'react'
import { FaHeart } from 'react-icons/fa'
import styled from 'styled-components/macro'

export default styled(FaHeart)`
  position: absolute;
  text-align: right;
  left: 88%;
  top: 12px;
  height: 32px;
  width: 32px;
  padding: 4px;
  background: var(--primary-backgroundopaque);
  border-radius: 4px;
  color: ${(props) => (props.isFavourite ? '#c82a1a' : 'white')};
  transition-duration: 0.8s;
`
