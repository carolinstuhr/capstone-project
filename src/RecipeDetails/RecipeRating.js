import React from 'react'
import styled from 'styled-components/macro'
import { AiOutlineStar } from 'react-icons/ai'

export default function RecipeRating({
  setIsRatingWindowVisible,
  recipeRating,
}) {
  return (
    <>
      <FavouritesSection onClick={() => setIsRatingWindowVisible(true)}>
        <StarIcon1 recipeRating={recipeRating} />
        <StarIcon2 recipeRating={recipeRating} />
        <StarIcon3 recipeRating={recipeRating} />
        <StarIcon4 recipeRating={recipeRating} />
        <StarIcon5 recipeRating={recipeRating} />
        <ParagraphStyled>ratings</ParagraphStyled>
        {console.log(recipeRating)}
      </FavouritesSection>
    </>
  )
}
const FavouritesSection = styled.section``

const StarIcon1 = styled(AiOutlineStar)`
  color: ${(props) => (props.recipeRating > 0 ? '#c82a1a' : 'white')};
`
const StarIcon2 = styled(AiOutlineStar)`
  color: ${(props) => (props.recipeRating > 1 ? '#c82a1a' : 'white')};
`
const StarIcon3 = styled(AiOutlineStar)`
  color: ${(props) => (props.recipeRating > 2 ? '#c82a1a' : 'white')};
`
const StarIcon4 = styled(AiOutlineStar)`
  color: ${(props) => (props.recipeRating > 3 ? '#c82a1a' : 'white')};
`
const StarIcon5 = styled(AiOutlineStar)`
  color: ${(props) => (props.recipeRating > 4 ? '#c82a1a' : 'white')};
`

const ParagraphStyled = styled.p`
  margin: 0;
  padding: 0;
  text-align: center;
  font-size: 12px;
`
