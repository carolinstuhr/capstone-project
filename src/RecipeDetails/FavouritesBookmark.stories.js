import React from 'react'
import FavouritesBookmark from './FavouritesBookmark'
import { action } from '@storybook/addon-actions'
import { withKnobs, color } from '@storybook/addon-knobs'

export default {
  component: FavouritesBookmark,
  title: 'FavouritesBookmark',
  decorators: [withKnobs],
}

export const ToggleHeart = () => (
  <FavouritesBookmark onClick={action('changed')} />
)

const label = 'Color'
const defaultValue = 'white'

export const FavouritesColor = () => (
  <FavouritesBookmark color={color(label, defaultValue)} />
)
