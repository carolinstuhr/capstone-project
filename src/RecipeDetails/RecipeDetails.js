import React, { useState } from 'react'
import styled from 'styled-components/macro'
import LeftArrow from '../images/left-arrow.svg'
import FavouritesBookmark from './FavouritesBookmark'
import { Link, useParams } from 'react-router-dom'
import { db } from '../firebaseConfig'

export default function RecipeDetails({ user, recipes, previousPage }) {
  const params = useParams()

  const [recipeDetails, setRecipeDetails] = useState('ingredients')

  let recipe = recipes && recipes.find((recipe) => recipe.id === params.id)

  const [isFavourite, setIsFavourite] = useState(
    user && user.favourites.some((favourite) => favourite === recipe.id)
  )
  let favouriteCheck =
    user && user.favourites.some((favourite) => favourite === recipe.id)

  return (
    <>
      {recipe && (
        <MainStyled>
          <ImageSectionStyled>
            {previousPage === 'All' && (
              <Link exact to="/">
                <ArrowIconStyled
                  src={LeftArrow}
                  alt="return to all recipes"
                  onClick={() => setRecipeDetails('ingredients')}
                  className="return-all"
                />
              </Link>
            )}
            {previousPage === 'Favourites' && (
              <Link exact to="/favourites">
                <ArrowIconStyled
                  src={LeftArrow}
                  alt="return to favourite recipes"
                  onClick={() => setRecipeDetails('ingredients')}
                  className="returnFavourites"
                />
              </Link>
            )}
            {previousPage === 'Profile' && (
              <Link to="/profile">
                <ArrowIconStyled
                  src={LeftArrow}
                  alt="return to profile page"
                  onClick={() => setRecipeDetails('ingredients')}
                  className="return-profile"
                />
              </Link>
            )}
            <FavouritesBookmark
              onClick={() => {
                toggleHeartIcon(recipe)
              }}
              isFavourite={isFavourite || favouriteCheck}
              alt="bookmark recipe"
              className="heart-icon"
            />
            <ImageStyled src={recipe.image} alt="" />
          </ImageSectionStyled>
          <RecipeInfoSectionStyled>
            <TitleStyled>{recipe.title}</TitleStyled>
            <InfoSection>
              <span>serves: {recipe.serving}</span>
              <span>
                time: {recipe.timehour}:{recipe.timeminutes}
              </span>
            </InfoSection>
            <DetailSelectionStyled>
              <IngredientsSelectionSpan
                onClick={() => setRecipeDetails('ingredients')}
                recipeDetails={recipeDetails}
                className="ingredients-selector"
              >
                Ingredients
              </IngredientsSelectionSpan>
              <InstructionsSelectionSpan
                onClick={() => setRecipeDetails('instructions')}
                recipeDetails={recipeDetails}
                data-testid="instructionsButton"
                className="instructions-selector"
              >
                Instructions
              </InstructionsSelectionSpan>
            </DetailSelectionStyled>
          </RecipeInfoSectionStyled>
          {recipeDetails === 'ingredients' ? (
            <IngredientsSection>
              {recipe.ingredients.map((ingredient, index) => (
                <>
                  <StyledParagraph key={index}>
                    {ingredient.amount}
                  </StyledParagraph>
                  <StyledParagraph>{ingredient.name}</StyledParagraph>
                </>
              ))}
            </IngredientsSection>
          ) : (
            <InstructionsSection>
              {recipe.instructions.map((instruction, index) => (
                <>
                  <StyledParagraph key={index}>{index + 1}.</StyledParagraph>
                  <StyledParagraph>{instruction}</StyledParagraph>
                </>
              ))}
            </InstructionsSection>
          )}
        </MainStyled>
      )}
    </>
  )
  function toggleHeartIcon(recipe) {
    let index = user.favourites.indexOf(recipe.id)
    index >= 0
      ? db
          .collection('users')
          .doc(user.id)
          .update({
            favourites: [
              ...user.favourites.slice(0, index),
              ...user.favourites.slice(index + 1),
            ],
          })
          .then(() => {
            setIsFavourite(false)
          })
          .catch((err) =>
            alert('Something went wrong. Please try again later.', err)
          )
      : db
          .collection('users')
          .doc(user.id)
          .update({ favourites: [...user.favourites, recipe.id] })
          .then(() => {
            setIsFavourite(true)
          })
          .catch((err) =>
            alert('Something went wrong. Please try again later.', err)
          )
  }
}
const MainStyled = styled.main`
  overflow-x: hidden;
  padding-top: 0;
`

const ImageSectionStyled = styled.section`
  height: 288px;
  position: relative;
`

const ArrowIconStyled = styled.img`
  background: var(--primary-backgroundopaque);
  border-radius: 4px;
  height: 32px;
  width: 32px;
  position: absolute;
  left: 12px;
  top: 12px;
  padding: 4px;
`

const ImageStyled = styled.img`
  height: 288px;
  width: 100%;
  object-fit: cover;
`

const RecipeInfoSectionStyled = styled.section`
  border-radius: 8px;
  position: relative;
  bottom: 12px;
  padding: 12px;
  background: var(--primary-background);
`

const TitleStyled = styled.h3`
  text-align: center;
  margin-top: 0;
  margin-bottom: 8px;
  text-transform: capitalize;
  font-weight: 400;
  font-size: 22px;
  animation-duration: 1s;
  animation-duration: 2s;
  animation-name: fadein;
  @keyframes fadein {
    from {
      opacity: 0;
    }

    to {
      opacity: 100;
    }
  }
`
const InfoSection = styled.section`
  display: flex;
  padding-left: 12px;
  padding-right: 12px;
  justify-content: space-between;
  margin-top: 16px;
  margin-bottom: 12px;
  font-weight: 300;
  animation-duration: 2s;
  animation-name: fadein;
  @keyframes fadein {
    from {
      opacity: 0;
    }

    to {
      opacity: 100;
    }
  }
`

const DetailSelectionStyled = styled.div`
  display: flex;
  justify-content: center;
  margin-top: 4px;
  font-weight: 300;
  animation-duration: 1s;
  animation-duration: 2s;
  animation-name: fadein;
  @keyframes fadein {
    from {
      opacity: 0;
    }

    to {
      opacity: 100;
    }
  }
`
const IngredientsSelectionSpan = styled.span`
  padding: 4px;
  border: 2px solid white;
  border-right-width: 1px;
  cursor: default;
  background: ${(props) =>
    props.recipeDetails === 'ingredients'
      ? 'var(--primary-background)'
      : 'rgba(255, 255, 255, 0.4)'};
  color: ${(props) =>
    props.recipeDetails === 'ingredients'
      ? 'var(--primary)'
      : 'var(--primary-opaque)'};
`

const InstructionsSelectionSpan = styled.span`
  padding: 4px;
  border: 2px solid white;
  border-left-width: 1px;
  cursor: default;
  background: ${(props) =>
    props.recipeDetails === 'instructions'
      ? 'var(--primary-background)'
      : 'rgba(255, 255, 255, 0.4)'};
  color: ${(props) =>
    props.recipeDetails === 'instructions'
      ? 'var(--primary)'
      : 'var(--primary-opaque)'};
`

const IngredientsSection = styled.section`
  padding-left: 12px;
  padding-right: 12px;
  display: grid;
  grid-template-columns: 1fr 3fr;
  font-weight: 300;
  animation-duration: 2s;
  animation-name: fadein;
  @keyframes fadein {
    from {
      opacity: 0;
    }

    to {
      opacity: 100;
    }
  }
`

const InstructionsSection = styled.section`
  padding-left: 12px;
  padding-right: 8px;
  display: grid;
  grid-template-columns: 1fr 20fr;
  font-weight: 300;
  animation-duration: 2s;
  animation-name: fadein;
  @keyframes fadein {
    from {
      opacity: 0;
    }

    to {
      opacity: 100;
    }
  }
`

const StyledParagraph = styled.p`
  margin-top: 4px;
`
