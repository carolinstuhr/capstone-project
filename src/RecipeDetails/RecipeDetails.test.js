import React from 'react'
import RecipeDetails from './RecipeDetails'
import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import recipeData from '../RecipeList.json'

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: () => ({ id: 1 }),
}))

test('renders page with recipe id 1', () => {
  const tree = render(
    <MemoryRouter>
      <RecipeDetails recipes={recipeData} />
    </MemoryRouter>
  )
  expect(tree).toMatchSnapshot()
})

test('renders title of recipe id 1', () => {
  const { getByText } = render(
    <MemoryRouter>
      <RecipeDetails recipes={recipeData} />
    </MemoryRouter>
  )
  expect(getByText('Chocolate chip cookies')).toBeInTheDocument()
})
