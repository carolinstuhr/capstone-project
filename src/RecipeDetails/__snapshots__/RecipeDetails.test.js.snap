// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders page with recipe id 1 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="RecipeDetails__MainStyled-sc-1urzn0d-0 dfKsYk"
      >
        <section
          class="RecipeDetails__ImageSectionStyled-sc-1urzn0d-1 bCFoki"
        >
          <svg
            alt="bookmark recipe"
            class="FavouritesBookmark-sc-18jcs1h-0 ihnwd heart-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
            />
          </svg>
          <img
            alt=""
            class="RecipeDetails__ImageStyled-sc-1urzn0d-2 hjcPyA"
            src="./images/chocolate_chip_cookies.jpg"
          />
        </section>
        <section
          class="RecipeDetails__RecipeInfoSectionStyled-sc-1urzn0d-3 dESenH"
        >
          <h3
            class="RecipeDetails__TitleStyled-sc-1urzn0d-4 kJqMjz"
          >
            Chocolate chip cookies
          </h3>
          <section
            class="RecipeDetails__InfoSection-sc-1urzn0d-5 dFyLfW"
          >
            <span>
              serves: 
              10
            </span>
            <section
              class="RecipeRating__FavouritesSection-sc-4qz39f-0 dDnGBJ"
            >
              <svg
                class="RecipeRating__StarIcon1-sc-4qz39f-1 BXRcB"
                fill="currentColor"
                height="1em"
                recipeRating="NaN"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                />
              </svg>
              <svg
                class="RecipeRating__StarIcon2-sc-4qz39f-2 lapzLF"
                fill="currentColor"
                height="1em"
                recipeRating="NaN"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                />
              </svg>
              <svg
                class="RecipeRating__StarIcon3-sc-4qz39f-3 bdryzF"
                fill="currentColor"
                height="1em"
                recipeRating="NaN"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                />
              </svg>
              <svg
                class="RecipeRating__StarIcon4-sc-4qz39f-4 hCyytx"
                fill="currentColor"
                height="1em"
                recipeRating="NaN"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                />
              </svg>
              <svg
                class="RecipeRating__StarIcon5-sc-4qz39f-5 iUWgGt"
                fill="currentColor"
                height="1em"
                recipeRating="NaN"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
                />
              </svg>
              <p
                class="RecipeRating__ParagraphStyled-sc-4qz39f-6 gNTUuM"
              >
                ratings
              </p>
            </section>
            <span>
              time: 
              0
              :
              40
            </span>
          </section>
          <div
            class="RecipeDetails__DetailSelectionStyled-sc-1urzn0d-6 iXWNBS"
          >
            <span
              class="RecipeDetails__IngredientsSelectionSpan-sc-1urzn0d-7 bVnKZl ingredients-selector"
            >
              Ingredients
            </span>
            <span
              class="RecipeDetails__InstructionsSelectionSpan-sc-1urzn0d-8 kmaLsl instructions-selector"
              data-testid="instructionsButton"
            >
              Instructions
            </span>
          </div>
        </section>
        <section
          class="RecipeDetails__IngredientsSection-sc-1urzn0d-9 dgqXzX"
        >
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            200 g
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            flour
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            100 ml
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            milk
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            250 g
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            sugar
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            2
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            eggs
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            1 tsp
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            baking powder
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            100 g
          </p>
          <p
            class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
          >
            chocolate chips
          </p>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="RecipeDetails__MainStyled-sc-1urzn0d-0 dfKsYk"
    >
      <section
        class="RecipeDetails__ImageSectionStyled-sc-1urzn0d-1 bCFoki"
      >
        <svg
          alt="bookmark recipe"
          class="FavouritesBookmark-sc-18jcs1h-0 ihnwd heart-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"
          />
        </svg>
        <img
          alt=""
          class="RecipeDetails__ImageStyled-sc-1urzn0d-2 hjcPyA"
          src="./images/chocolate_chip_cookies.jpg"
        />
      </section>
      <section
        class="RecipeDetails__RecipeInfoSectionStyled-sc-1urzn0d-3 dESenH"
      >
        <h3
          class="RecipeDetails__TitleStyled-sc-1urzn0d-4 kJqMjz"
        >
          Chocolate chip cookies
        </h3>
        <section
          class="RecipeDetails__InfoSection-sc-1urzn0d-5 dFyLfW"
        >
          <span>
            serves: 
            10
          </span>
          <section
            class="RecipeRating__FavouritesSection-sc-4qz39f-0 dDnGBJ"
          >
            <svg
              class="RecipeRating__StarIcon1-sc-4qz39f-1 BXRcB"
              fill="currentColor"
              height="1em"
              recipeRating="NaN"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
              />
            </svg>
            <svg
              class="RecipeRating__StarIcon2-sc-4qz39f-2 lapzLF"
              fill="currentColor"
              height="1em"
              recipeRating="NaN"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
              />
            </svg>
            <svg
              class="RecipeRating__StarIcon3-sc-4qz39f-3 bdryzF"
              fill="currentColor"
              height="1em"
              recipeRating="NaN"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
              />
            </svg>
            <svg
              class="RecipeRating__StarIcon4-sc-4qz39f-4 hCyytx"
              fill="currentColor"
              height="1em"
              recipeRating="NaN"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
              />
            </svg>
            <svg
              class="RecipeRating__StarIcon5-sc-4qz39f-5 iUWgGt"
              fill="currentColor"
              height="1em"
              recipeRating="NaN"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3zM664.8 561.6l36.1 210.3L512 672.7 323.1 772l36.1-210.3-152.8-149L417.6 382 512 190.7 606.4 382l211.2 30.7-152.8 148.9z"
              />
            </svg>
            <p
              class="RecipeRating__ParagraphStyled-sc-4qz39f-6 gNTUuM"
            >
              ratings
            </p>
          </section>
          <span>
            time: 
            0
            :
            40
          </span>
        </section>
        <div
          class="RecipeDetails__DetailSelectionStyled-sc-1urzn0d-6 iXWNBS"
        >
          <span
            class="RecipeDetails__IngredientsSelectionSpan-sc-1urzn0d-7 bVnKZl ingredients-selector"
          >
            Ingredients
          </span>
          <span
            class="RecipeDetails__InstructionsSelectionSpan-sc-1urzn0d-8 kmaLsl instructions-selector"
            data-testid="instructionsButton"
          >
            Instructions
          </span>
        </div>
      </section>
      <section
        class="RecipeDetails__IngredientsSection-sc-1urzn0d-9 dgqXzX"
      >
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          200 g
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          flour
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          100 ml
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          milk
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          250 g
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          sugar
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          2
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          eggs
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          1 tsp
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          baking powder
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          100 g
        </p>
        <p
          class="RecipeDetails__ParagraphStyled-sc-1urzn0d-11 dWkeEu"
        >
          chocolate chips
        </p>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
